Annotation filter doc

soyuka opened the pull request:
@meyerbaptiste where are the docs for Property and Group filters?

meyerbaptiste commented:
Here @soyuka: https://github.com/api-platform/docs/blob/2.1/core/filters.md#serializer-filters

Why? 😃

soyuka commented:
I'm stupid, mb I think I didn't based my commit on the origin/master.

bendavies commented:
@soyuka think you committed from using the wrong git user

soyuka commented:
I know, wasn't my computer and lazy to git config on a temporary hardware. Doesn't change anything as I'm waiting for reviews and will probably edit/rebase with my correct one.

soyuka commented:
WDYT?

soyuka commented:
I've updated the filters doc, may you take a look?

soyuka commented on core/filters.md:
```diff
@@ -285,27 +284,20 @@ Syntax: `?order[property]=<asc|desc>`
 /**
- * @ApiResource(attributes={"filters"={"offer.order_filter"}})
+ * @ApiResource
+ * @ApiFilter(OrderFilter::class, properties={"id", "name"})
```
I need to review this as this will not work yet.

sroze commented:
It's conflicting now 🤷‍♂️

soyuka commented:
Yes because I changed the whole docs for annotation filters :). Should I rebase?

dunglas commented:
Can you rebase @soyuka? Feel free to merge it :)

soyuka commented:
It's a mess to rebase oO

soyuka commented:
Because it's a lot of work to rebase I'm merging this so that next filter changes occurs on the new version. Because it's master I think it's okay if not we can still revert :).  

Done:
- changed every examples to use the `ApiFilter` annotation
- added back the note on how the annotation works
- simplified Property, Group filter options enumeration
